test/Test/Data/Array.purs

```diff
@@ -139,6 +147,14 @@ testArray = do
   assert $ (A.elemLastIndex 1 [1, 2, 1]) == Just 2
   assert $ (A.elemLastIndex 4 [1, 2, 1]) == Nothing
 
+  log "find should return the first element for which a predicate returns true in an array"
+  assert $ (A.find (_ /= 1) [1, 2, 1]) == Just 2
+  assert $ (A.find (_ == 3) [1, 2, 1]) == Nothing
+
+  log "findMap should return the mapping of the first element that satisfies the given predicate"
```

Comment:
I think it would be good to also add a test case for an array where there is more than one element which satisfies the predicate here.

Reply:
Fixed in latest commits

Reply:
Sorry, I should have been clearer: I think we should have a test which rules out the possibility that our `findMap` actually returns the _last_ match.

Reply:
I've updated the tests to account for that.